ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-settings/query-settings.view.js

```diff
@@ -93,11 +97,15 @@ module.exports = plugin(
       })
       resultTemplates = _.uniq(resultTemplates, 'id')
       let lastIndex = resultTemplates.length - 1
+      let defaultResultForm = resultTemplates.find(
+        form => form.id === user.getQuerySettings().get('defaultResultFormId')
+      )
       let detailLevelProperty = new Property({
         label: 'Result Form',
         enum: resultTemplates,
         value: [
-          this.model.get('detail-level') ||
+          (defaultResultForm && defaultResultForm.value) ||
+            this.model.get('detail-level') ||
```

Comment:
How do `defaultResultFormId` and `detail-level` now relate to each other?

Reply:
the `detail-level` isn't the id, it's another unique identifier. It can have the value `allFields` which the `defaultResultFormId` cannot. I could try replacing `this.model.get('detail-level')` with `allFields` - i think this would achieve the desired logic

Reply:
I'm still confused as to what the difference is between the two attributes are. I understand that they reference result forms in different ways (one by id and the other by name), but they both represent a default result form and the distinction is unclear to me

Reply:
I think the `detail-level` could have a result form without it being the "default" result form. Like, if aa user selects a result form while on a search form, that result form would be the `detail-level`. but the default result form is something that's set in the result form selection, and isn't necessarily the search form's result form. 

In other words.. search form A could have result form X, and you've made search from A your default search form. Then, you decide to make result form Y your default result form. When you go to search, it will default to search form A, which will show X rather than Y. In this case, X is the `detail-level` and Y is the `defaultResultFormId`